### PR TITLE
Refactored Kata

### PR DESCRIPTION
**Pull request by 2unes**
Tell me what else that I need to Refactor

### CODE REVIEW

File: RefactorKata/RefactorKata/Program.cs [hunk 1]
```diff
@@ -1,41 +1,44 @@
...
 {
-    class Program
+    internal class Program
     {
-        static void Main(string[] args)
+        private static void Main(string[] args)
         {
-            //This is intentionally bad : (  Let's Refactor!
-            System.Data.SqlClient.SqlConnection Conn = new System.Data.SqlClient.SqlConnection("Server=.;Database=myDataBase;User Id=myUsername;Password = myPassword;");
+            var conn = new SqlConnection("Server=.;Database=myDataBase;User Id=myUsername;Password = myPassword;");
```
Comment (soitgoes) [inline RefactorKata/RefactorKata/Program.cs]: Let's integrate the "using" keyword.  Also let's extract the connection string to App.config and reference that with ConnectionManager.ConnectionStrings

File: RefactorKata/RefactorKata/Program.cs [hunk 2]
```diff
@@ -1,41 +1,44 @@
...
             }
-            Conn.Dispose();
             Console.WriteLine("Products Loaded!");
-            for (int i =0; i< products.Count; i++)
+            conn.Dispose();
+
+            foreach (var t in products)
             {
-                Console.WriteLine(products[i].name);
+                Console.WriteLine(t.name);
```
Comment (soitgoes) [inline RefactorKata/RefactorKata/Program.cs]: let's use a more descriptive variable name than "t"

File: RefactorKata/RefactorKata/Program.cs [hunk 3]
```diff
@@ -1,41 +1,44 @@
...
     {
-        static void Main(string[] args)
+        private static void Main(string[] args)
         {
-            //This is intentionally bad : (  Let's Refactor!
-            System.Data.SqlClient.SqlConnection Conn = new System.Data.SqlClient.SqlConnection("Server=.;Database=myDataBase;User Id=myUsername;Password = myPassword;");
+            var conn = new SqlConnection("Server=.;Database=myDataBase;User Id=myUsername;Password = myPassword;");
 
-            System.Data.SqlClient.SqlCommand cmd = Conn.CreateCommand();
+            var cmd = conn.CreateCommand();
```
Comment (soitgoes) [inline RefactorKata/RefactorKata/Program.cs]: Would like to see a GetProducts method extracted out.  Right now we just have one big method.

File: RefactorKata/RefactorKata/Program.cs [hunk 4]
```diff
@@ -1,41 +1,44 @@
...
             {
-                Console.WriteLine(products[i].name);
+                Console.WriteLine(t.name);
             }
         }
     }
     public class Product
     {
-        public string name;
-        public string Name { get { return name; } set { name = value; } }
```
Comment (soitgoes) [inline RefactorKata/RefactorKata/Program.cs]: Would like to see Product in it's own file.

**Comment by 2unes** [comment]
Re-submitted

File: RefactorKata/RefactorKata/Program.cs [hunk 5]
```diff
@@ -1,41 +1,44 @@
...
-                prod.name = reader["Name"].ToString();
-                products.Add(prod);
+                Console.WriteLine(Products.name);
             }
-            Conn.Dispose();
-            Console.WriteLine("Products Loaded!");
-            for (int i =0; i< products.Count; i++)
+        }
+
+        private static List<Product> GetProducts()
```
Comment (soitgoes) [inline RefactorKata/RefactorKata/Program.cs]: Would like comment doucmentation on the method. "///"

File: RefactorKata/RefactorKata/Program.cs [hunk 6]
```diff
@@ -1,41 +1,44 @@
...
+        {
+            List<Product> products;
+            using (var conn = new SqlConnection("Server=.;Database=myDataBase;User Id=myUsername;Password = myPassword;"))
```
Comment (soitgoes) [inline RefactorKata/RefactorKata/Program.cs]: Would still like to see connection string pulled out to App.config.  Will review tomorrow.

**Pull request closed without merging**